spring-integration-kafka/src/main/java/org/springframework/integration/kafka/inbound/KafkaInboundGateway.java

```diff
@@ -192,9 +204,6 @@ protected void onInit() {
 			}
 		}
 		ContainerProperties containerProperties = this.messageListenerContainer.getContainerProperties();
-		Object existing = containerProperties.getMessageListener();
-		Assert.state(existing == null, () -> "listener container cannot have an existing message listener (" + existing
-					+ ")");
```

Comment:
Why remove these assertions? You asked me to add them last week.

Reply:
Sorry for confusion: we have had them before in the ctor anyway.
I guess I shouldn't do this since it might cause a conflict on cherry-pick...